### PR TITLE
Add `proximityPrecision` management methods

### PR DESCRIPTION
**Pull request by brunoocasali**
Pre-release work for v1.6.

### CODE REVIEW

File: tests/Settings/ProximityPrecisionTest.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        parent::setUp();
+        $this->index = $this->createEmptyIndex($this->safeIndexName());
+    }
+
+    public function testGetDefaultProximityPrecision(): void
+    {
+        $default = $this->index->getProximityPrecision();
+
+        $this->assertEquals($default, "byWord");
```
Comment (norkunas) [inline tests/Settings/ProximityPrecisionTest.php:24]:
```suggestion
        $this->assertSame('byWord', $default);
```

File: tests/Settings/ProximityPrecisionTest.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public function testUpdateProximityPrecision(): void
+    {
+        $promise = $this->index->updateProximityPrecision("byAttribute");
+        $this->assertIsValidPromise($promise);
+        $this->index->waitForTask($promise['taskUid']);
+
+        $this->assertEquals("byAttribute", $this->index->getProximityPrecision());
```
Comment (norkunas) [inline tests/Settings/ProximityPrecisionTest.php:33]:
```suggestion
        $this->assertSame('byAttribute', $this->index->getProximityPrecision());
```

File: tests/Settings/ProximityPrecisionTest.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        $promise = $this->index->updateProximityPrecision("byAttribute");
+        $this->assertIsValidPromise($promise);
+        $this->index->waitForTask($promise['taskUid']);
+
+        $promise = $this->index->resetProximityPrecision();
+
+        $this->assertIsValidPromise($promise);
+        $this->index->waitForTask($promise['taskUid']);
+
+        $this->assertEquals($this->index->getProximityPrecision(), "byWord");
```
Comment (norkunas) [inline tests/Settings/ProximityPrecisionTest.php:47]:
```suggestion
        $this->assertEquals('byWord', $this->index->getProximityPrecision());
```

**Review by norkunas — CHANGES_REQUESTED** [review]

File: .code-samples.meilisearch.yaml [hunk 4]
```diff
@@ -697,3 +697,9 @@ update_non_separator_tokens_1: |-
   $client->index('articles')->updateNonSeparatorTokens(['@', '#']);
 reset_non_separator_tokens_1: |-
   $client->index('articles')->resetNonSeparatorTokens();
+get_proximity_precision_settings_1: |-
+  $client->index('books')->getProximityPrecision();
+update_proximity_precision_settings_1: |-
+  $client->index('books')->updateProximityPrecision("byAttribute");
```
Comment (norkunas) [inline .code-samples.meilisearch.yaml:703]:
```suggestion
  $client->index('books')->updateProximityPrecision('byAttribute');
```

**Review by norkunas — CHANGES_REQUESTED** [review]

**Review by norkunas — APPROVED** [review]

**Review by curquiza — CHANGES_REQUESTED** [review]
Thanks Bruno

Can you fix PHP stan CI?

File: src/Endpoints/Delegates/HandlesSettings.php [hunk 5]
```diff
@@ -352,18 +352,12 @@ public function getProximityPrecision(): string
         return $this->http->get(self::PATH.'/'.$this->uid.'/settings/proximity-precision');
     }
 
-    /**
-     * @param non-empty-string $type
-     */
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesSettings.php]: why removed this one?

Comment (norkunas) [reply src/Endpoints/Delegates/HandlesSettings.php]: also I wonder is it an enum or can be anything?

**Pull request merged**